lib/dialects/postgres/query-generator.js

```diff
@@ -725,16 +725,23 @@ var QueryGenerator = {
 
     var enumName = '"enum_' + (_.isPlainObject(tableName) ? tableName.table : tableName) + '_' + attr + '"';
 
-    if (options.schema !== false && tableName && tableName.schema) {
-      enumName = '"' + tableName.schema + '"' + tableName.delimiter + enumName;
+    if (options.schema !== false && (tableName.schema || options.schema)) {
+      var schema = tableName.schema || options.schema || 'public';
+      var delimiter = tableName.delimiter || options.delimiter || '.';
```

Comment:
This code handles having schema/delimter be part of the table name and be part of the options.
I imagine we'll need this multiple places since a lot of queryInterface (migration) calls should support both, ideally we could provide some helper that takes a tableName and an options and returns schema, delimiter and tableName.